dev/tests/js/framework/JsTestRunner.php

```diff
@@ -0,0 +1,537 @@
+<?php
+
+class JsTestRunner
```

Comment:
As stated in the commit message I didn't see this as the final result, it just was some refactoring along the way, to leave the code in a better state then it was. Decomposing the class would probably be one of the next steps.
For now it got me where I wanted, that is, understand what the code does and make future changes easier.

I actually don't expect this code to hang around for long. Based on @vkorotun's comments in #954 I expect the JS testing framework will change quite a lot, maybe completely removing JsTestRunner (I hope so).